scripts/bake-ide-ami.ps1

```diff
@@ -672,29 +672,63 @@ $jsonObject = @"
             # Ensure last exit code is 0. (exit by itself will terminate the remote session)
             cmd /c exit 0
         }
-
         # Load up some required tools into remote environment
 
         # Load basic utils before running anything
         Execute-RemoteScript -Session $Script:session -FilePath "$script:IncludeDir\dot-CommonTools.ps1"
 
 
         if ( $InstallBaseSoftware ) {
-
+            
             # Install Chocolatey
-
             Execute-RemoteScript -Session $Script:session -FilePath "$script:IncludeDir\getchoco.ps1"
 
-            # Then we install git using chocolatey and pull down the rest of the files from git
+            if ( $Cloud -eq 'Azure' ) {
+                Write-Host("$(Log-Date) Azure requires a reboot after installing choco. Rebooting now..")
+                
+                Execute-RemoteBlock $Script:session {
+                    Restart-Computer -force
+                }
+                Start-Sleep -Seconds 10
+            
+                Write-Host "$(Log-Date) Reconnecting session..."
+                if ( $Script:session ) { Remove-PSSession $Script:session | Out-Default | Write-Host }
+            
+                Connect-RemoteSession | Out-Default | Write-Host
+                Execute-RemoteInit | Out-Default | Write-Host
+                Execute-RemoteScript -Session $Script:session -FilePath "$script:IncludeDir\dot-CommonTools.ps1"
+                Write-host "$(Log-Date) Rebooted!"
+            }
+            
 
+            # Then we install git using chocolatey and pull down the rest of the files from git
+            Write-Host "Installing Git!"
+            if ( $Cloud -eq 'Azure' ) {
+                
+                Invoke-Command -Session $Script:session {Set-ExecutionPolicy Unrestricted -Scope CurrentUser}
+                $remotelastexitcode = invoke-command  -Session $Script:session -ScriptBlock { $lastexitcode}
+                if ( $remotelastexitcode -and $remotelastexitcode -ne 0 ) {
+                    Write-Error "LastExitCode: $remotelastexitcode"
+                    throw 1
+                }
+            }
```

Comment:
This block of code to set the execution policy needs to occur before Execute-RemoteInit. Like it does on about line 625.